src/sort/index.ts

```diff
@@ -61,15 +62,20 @@ class Sort {
     }
   }
 
-  getPastPurchasesLabel(sort: Store.Sort) {
-    return sort.field;
+  getPastPurchasesLabel(sort: Store.Sort, index: number) {
+    if (index < this.props.pastPurchasesLabels.length) {
+      return this.props.pastPurchasesLabels[index];
+    } else {
+      return `${sort.field} ${sort.descending ? 'Descending' : 'Ascending'}`;
+    }
   }
 }
 
 interface Sort extends Tag<Sort.Props, Sort.State> {}
 namespace Sort {
   export interface Props extends Tag.Props {
     labels: string[];
+    pastPurchasesLabels: string[];
```

Comment:
Just noticing that the prop definition doesn't include `storeSection` please add it